main/autogen/autogen.h

```diff
@@ -81,13 +85,22 @@ struct ParsedFile {
 
     std::string toString(core::Context ctx);
     std::string toMsgpack(core::Context ctx, int version);
-    void classlist(core::Context ctx, std::vector<std::string> &out);
+    std::vector<std::string> listAllClasses(const core::Context ctx);
+    std::optional<AutogenSubclassMap> listAllSubclasses(const core::Context ctx,
+                                                        const std::vector<std::string> &absolutePathsToIgnore,
+                                                        const std::vector<std::string> &relativePathsToIgnore);
 
 private:
     std::vector<core::NameRef> showFullName(core::Context ctx, DefinitionRef id);
     friend class MsgpackWriter;
 };
 
+void maybeInsertChild(const std::string &parentName, const AutogenSubclassSet &children, AutogenSubclassMap &out);
+void patchChildMap(AutogenSubclassMap &childMap);
+void descendantsOf(const AutogenSubclassMap &childMap, const std::string &parent, AutogenSubclassSet &out);
+std::unique_ptr<std::vector<std::string>> serializeSubclassMap(const AutogenSubclassMap &descendantsMap,
+                                                               const std::vector<std::string> &parentNames);
+
```

Comment:
does it make sense to separate `autogen-sublcasses` from `autogen`?
Do they have to be in the same `h` file and same `cc` file?

Reply:
Done in https://github.com/sorbet/sorbet/pull/1040/commits/88c5a1d5b21424b25cf620f0288c3cd167aea8fe